Add 'solo5-mkimage' tool.

mato opened the pull request:
solo5-mkimage can be used to build MBR partitioned disk images
containing a unikernel. Includes support for GCE-style images (sparse
.tar.gz).

Only runs on Linux and/or in Docker, mainly due to lack of portable DOS
fdisk.

mato commented:
@ricarkol @sgrove Can you test this when you get a chance and confirm it works for you? 
@djwillia Ok to remove the existing solo5-build-iso.bash in favour of solo5-mkimage?

ricarkol commented on tools/mkimage/solo5-mkimage.sh:
```diff
@@ -0,0 +1,195 @@
+    SRCFILE=$(basename ${UNIKERNEL})
+
+    exec docker run --rm \
+        --tmpfs /tmp:rw,noexec,nosuid,size=2g \
```
Old versions of docker (e.g. mine) don't like this --tmpfs arg.

ricarkol commented on tools/mkimage/Dockerfile:
```diff
@@ -0,0 +1,11 @@
+FROM alpine:latest
```
Slight preference to use a specific alpine version. Just in case things change on the alpine side and something breaks mysteriously.

mato commented:
@ricarkol I've addressed all your comments. Shall I remove `solo5-build-iso.bash`? We can always get it back from history/integrate that functionality into `solo5-mkimage` if a use-case for ISO images is found.

ricarkol commented:
Thanks for the changes @mato
